rfcs/text/0010_service_status.md

```diff
@@ -0,0 +1,337 @@
+- Start Date: 2020-03-07
+- RFC PR: https://github.com/elastic/kibana/pull/59621
+- Kibana Issue: https://github.com/elastic/kibana/issues/41983
+
+# Summary
+
+A set API for describing the current status of a system (Core service or plugin)
+in Kibana.
+
+# Basic example
+
+```ts
+// Override default behavior and only elevate severity when elasticsearch is not available
+core.status.set(
+  core.status.core$.pipe(core => core.elasticsearch);
+)
+```
+
+# Motivation
+
+Kibana should do as much possible to help users keep their installation in a working state. This includes providing as much detail about components that are not working as well as ensuring that failures in one part of the application do not block using other portions of the application.
+
+In order to provide the user with as much detail as possible about any systems that are not working correctly, the status mechanism should provide excellent defaults in terms of expressing relationships between services and presenting detailed information to the user.
+
+# Detailed design
+
+## Types
+
+```ts
+/**
+ * The current status of a service at a point in time.
+ */
+type ServiceStatus = {
+  /**
+   * The current availability level of the service.
+   */
+  level: ServiceStatusLevel.available;
+  /**
+   * A high-level summary of the service status.
+   */
+  summary?: string;
+  /**
+   * A more detailed description of the service status.
+   */
+  detail?: string;
+  /**
+   * A URL to open in a new tab about how to resolve or troubleshoot the problem.
+   */
+  documentationUrl?: string;
```

Comment:
Great idea with documentation url. We can really improve in helping users understand what the problem is, and how to solve it. "Kibana server is not ready yet" 🤷‍♂ comes to mind...

Reply:
`Kibana server is not ready yet` is an interesting one too. I haven't thought through how this RFC could or should affect that functionality.

Reply:
> I haven't thought through how this RFC could or should affect that functionality.

Was probably a bad example anyway :) 
That specific exampe just happened to be on my mind after I saw [it was the most visited issue](https://github.com/elastic/kibana/issues/25806) in the entire repo (yeah I sometimes randomly look at the traffic insights - it's a weird hobby :p ).

Reply:
Interesting to know. Points to the fact that we should probably have more information in that page. Will consider that outside the scope of this RFC + project 👍